shader/fine.wgsl

```diff
@@ -265,6 +286,28 @@ fn main(
                 }
                 cmd_ix += 3u;
             }
+            // CMD_IMAGE
+            case 8u: {
+                let image = read_image(cmd_ix);
+                let atlas_extents = image.atlas_offset + image.extents;
+                for (var i = 0u; i < PIXELS_PER_THREAD; i += 1u) {
+                    let my_xy = vec2(xy.x + f32(i), xy.y);
+                    let atlas_uv = image.matrx.xy * my_xy.x + image.matrx.zw * my_xy.y + image.xlat + image.atlas_offset;
+                    // This currently clips to the image bounds. TODO: extend modes
+                    if all(atlas_uv < atlas_extents) {
+                        let uv_quad = vec4(max(floor(atlas_uv), image.atlas_offset), min(ceil(atlas_uv), atlas_extents));
+                        let uv_frac = fract(atlas_uv);
+                        let a = premul_alpha(textureLoad(image_atlas, vec2<i32>(uv_quad.xy), 0));
```

Comment:
I'm curious why the by-hand bilinear interpolation here rather than using hardware. Going over the WGSL intrinsics, I think `textureSampleLevel` is the most appropriate version - unlike `textureSample` it doesn't rely on derivatives, and I think it's likely to be more performant than `textureSampleGrad` (which based on past piet-gpu experimentation was slow).

That could perhaps be followup work to this patch.

Reply:
I hadn’t intended to do filtering at all in this PR so it was sort of tacked on as an afterthought. I’ll work on hardware sampling as a followup.